axelor-stock/src/main/java/com/axelor/apps/stock/service/TrackingNumberService.java

```diff
@@ -134,6 +137,15 @@ public TrackingNumber createTrackingNumber(
     }
     trackingNumber.setTrackingNumberSeq(seq);
 
+    // In case of barcode generation, retrieve the one set on tracking number configuration
+    AppStock appStock = Beans.get(AppStockService.class).getAppStock();
```

Comment:
please avoid using conditional injection for services and inject from the constructor